rtengine/iplocallab.cc

```diff
@@ -2388,9 +2391,10 @@ void ImProcFunctions::exlabLocal(local_params& lp, int bfh, int bfw, LabImage* b
         for (int jr = 0; jr < bfw; jr++) {
             float L = bufexporig->L[ir][jr];
 
-            if (L < mean && lp.expmet == 1 && lp.linear > 0.f && lp.laplacexp > 0.1f && !lp.invex) {
+        //    if (L < mean && lp.expmet == 1 && lp.linear > 0.f && lp.laplacexp > 0.1f && !lp.invex) {//disabled generate artifacts
+            if (lp.expmet == 1 && lp.linear > 0.f && lp.laplacexp > 0.1f && !lp.invex) {
                 float Llin = LIM01(L / 32768.f);
-                addcomp = linear * (-kl * Llin + kl);//maximum about 1.5 IL
+                addcomp = linear * (-kl * Llin + kl);//maximum about 1. IL
                 exp_scale = pow(2.0, (lp.expcomp + addcomp));
```

Comment:
This cries for `pow_F` for faster processing ;-)

Reply:
Oh, even worse, addcomp is declared outside the parallel loop, means we have a race condition here.
I will have a look at that this evening.

Reply:
Ok thank you

jacques

Reply:
@Desmis Jacques, can you please post a .pp3 to test this case?